### PR TITLE
Fix Brave Ads state-level targeting crash

### LINKED ISSUE #2692: Fix Brave Ads state-level targeting crash
### Description: 
Fix Brave Ads state-level targeting crash

### Steps to Reproduce 
  1. Disable state-level targeting in Ads settings (US Only)

**Actual result:** 
App crashes

**Expected result:**
App should not crash

**Reproduces how often:** [Easily reproduced, Intermittent Issue]
Easily reproduced

**Brave Version:** 
1.20.x

**Device details:** 

**Website problems only:**
- did you check with Brave Shields down?
- did you check in Safari/Firefox (WkWebView-based browsers)?

### PR DESCRIPTION
**Pull request by tmancey**
## Summary of Changes

This pull request fixes #2692 

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`

## Reviewer Checklist:

- [ ] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `release-notes/(include|exclude)`
  - `bug` / `enhancement`
- [ ] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [ ] Adequate unit test coverage exists to prevent regressions.
- [ ] Adequate test plan exists for QA to validate (if applicable).
- [ ] Issue is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: BraveRewardsUI/Settings/Ads/Details/AdsDetailsViewController.swift [hunk 1]
```diff
@@ -228,16 +228,18 @@ extension AdsDetailsViewController: UITableViewDelegate, UITableViewDataSource {
...
+      if state.ads.subdivisionTargetingCode == "AUTO" {
         adsSubdivisionTargetingCode = state.ads.automaticallyDetectedSubdivisionTargetingCode
       } else {
         adsSubdivisionTargetingCode = state.ads.subdivisionTargetingCode
       }
 
-      guard let selectedIndex = subdivisionTargetingOptions.firstIndex(where: { $0.0 == adsSubdivisionTargetingCode }) else { fatalError() }
-      cell.accessoryLabel?.text = subdivisionTargetingOptions[selectedIndex].1
+      if adsSubdivisionTargetingCode == "DISABLED" {
+        cell.accessoryLabel?.text = "Disabled"
```
Comment (kylehickinson) [inline BraveRewardsUI/Settings/Ads/Details/AdsDetailsViewController.swift]: This has to be localized

**Review by kylehickinson — CHANGES_REQUESTED** [review]

**Review by kylehickinson — CHANGES_REQUESTED** [review]
Still needs to be localized (put static text in entry in Strings.swift, then use that there)

**Comment by tmancey** [comment]
@kylehickinson Fixed localization issues

File: BraveRewardsUI/Settings/Ads/Details/AdsDetailsViewController.swift (hunk 1 above)
Comment (tmancey) [reply BraveRewardsUI/Settings/Ads/Details/AdsDetailsViewController.swift]: Fixed

**Review by kylehickinson — APPROVED** [review]

**Pull request merged**